x/ccv/provider/keeper/grpc_query.go

```diff
@@ -319,6 +320,10 @@ func (k Keeper) QueryConsumerValidators(goCtx context.Context, req *types.QueryC
 
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
+	if !k.IsConsumerProposedOrRegistered(ctx, consumerChainID) {
```

Comment:
will this not already be taken care of by the check on line 327? could you maybe add a comment why this check is needed?

Reply:
Thanks for pointing that out. You’re right—I made a mistake. The check on line 327 should handle this case, so the extra check isn’t needed.